src/construction.cpp

```diff
@@ -1241,6 +1243,39 @@ bool construct::check_support( const tripoint_bub_ms &p )
     return num_supports >= 2;
 }
 
+bool construct::check_support_below( const tripoint_bub_ms &p )
+{
+    bool blocking_creature = g->get_creature_if( [&]( const Creature & creature ) {
+        return creature.pos() == p.raw();
+    } ) != nullptr;
+
+    map &here = get_map();
+    // These checks are based on check_empty_lite, but accept no floor and the traps associated
+    // with it, i.e. ledge, pit, spiked pit, and glass pit.
+    // - Check if there's nothing in the way. The "passable" check rejects tiles you can't
+    //   pass through, but we want air and water to be OK as well (that's what we want to bridge).
+    // - Apart from that, vehicles, items, creatures, and furniture also have to be absent.
+    // - Then we have traps, and, unfortunately, there are "ledge" traps on all the open
+    //   space tiles adjacent to passable tiles, so we can't just reject all traps outright,
+    //   but have to accept those.
+    if( !( here.passable( p.raw() ) || here.has_flag( ter_furn_flag::TFLAG_LIQUID, p ) ||
+           here.has_flag( ter_furn_flag::TFLAG_NO_FLOOR, p ) ) ||
+        blocking_creature || here.has_furn( p ) || !( here.tr_at( p ).is_null() ||
+                here.tr_at( p ).id == tr_ledge  || here.tr_at( p ).id == tr_pit ||
+                here.tr_at( p ).name() == "spiked pit" || here.tr_at( p ).name() == "glass pit" ) ||
```

Comment:
Unless I misunderstood something, spiked pit has `tr_spike_pit` id and glass pit has `tr_glass_pit` id.

Also, instead of hardcoding, we could add a new trap flag like `PIT`.

Reply:
Yes, they have those ids (or close to them), but those are not among the ones hard coded, and so are not recognized when testing against "id". It's quite possible there's some way to get the string version of the id and compare against that, but I didn't see it.
Adding a trap flag would of course bypass the problem and make its easier to add additional pit trap variants without this part of the code being brittle. I'll take a look and report back.

Reply:
OK, a new flag seems to work, according to the tests.